Commons/src/main/java/tw/waterball/judgegirl/commons/utils/StreamUtils.java

```diff
@@ -73,6 +73,16 @@
         }).collect(toList());
     }
 
+    public static <T, R> List<T> mapOptionalToList(Collection<R> collection, ErrFunction<R, Optional<T>> mapping) {
+        return collection.stream().map(in -> {
+            try {
+                return mapping.apply(in).orElse(null);
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }).collect(toList());
+    }
+
```

Comment:
Remove this method,  change to  the ```flatMapToList``` method, it will help you to filter empty optional that you don't need it.